Client/qtTeamTalk/mainwindow.cpp

```diff
@@ -5410,10 +5410,26 @@ void MainWindow::slotFilesContextMenu(const QPoint &/* pos*/)
 {
     QMenu menu(this);
     QMenu* sortMenu = menu.addMenu(tr("Sort By..."));
-    QAction* sortName = sortMenu->addAction(tr("&Name"));
-    QAction* sortSize = sortMenu->addAction(tr("&Size"));
-    QAction* sortOwner = sortMenu->addAction(tr("&Owner"));
-    QAction* sortUpload = sortMenu->addAction(tr("&Upload Date"));
+    QAction* sortName = new QAction();
+    sortName->setText(tr("&Name"));
+    sortName->setCheckable(true);
+    sortName->setChecked((ttSettings->value(SETTINGS_DISPLAY_FILESLIST_SORT, SETTINGS_DISPLAY_FILESLIST_SORT_DEFAULT).toString() == "name")?true:false);

```

Comment:
The mapping from "name" in line 5416 to "name" in line 5446 is a bit unclear. It would be better to use a "const string" to emphasise the mapping. I.e.:

```
const QString name = "name";
sortName->setChecked((ttSettings->value(SETTINGS_DISPLAY_FILESLIST_SORT, SETTINGS_DISPLAY_FILESLIST_SORT_DEFAULT).toString() == name)?true:false);
...
ttSettings->setValue(SETTINGS_DISPLAY_FILESLIST_SORT, name);
```